src/game.cpp

```diff
@@ -578,54 +629,193 @@ void test_game() //!OCLINT tests may be many
+  /// An passive player cannot grow
   {
     game g;
 
-    //Make the first player invulnerable
-    become_invulnerable(g.get_player(0));
+    // Make the first player passive
+    g.set_player_state_passive(g.get_player(0));
 
     // Make player 1 and 2 overlap
     g.get_player(1).set_x(get_x(g.get_player(0)));
     g.get_player(1).set_y(get_y(g.get_player(0)));
-    assert(has_collision(g));
 
-    // After a tick, invulnerable player does not shrink
-    const int inv_player_size_before =  get_nth_player_size(g, 0);
+    const int first_player_index = get_collision_members(g)[0];
+    const int second_player_index = get_collision_members(g)[1];
+    const int winner_index = get_winning_player_index(g, first_player_index, second_player_index);
+
+    assert(winner_index == 0);
+
+    // After a tick, passive player does not grow
+    const int inv_player_size_before =  get_nth_player_size(g, winner_index);
     g.tick();
-    const int inv_player_size_after =  get_nth_player_size(g, 0);
+    const int inv_player_size_after =  get_nth_player_size(g, winner_index);
+    assert(inv_player_size_after == inv_player_size_before);
+  }
+
+  /// An passive player cannot shrink
+  {
+    game g;
+
+    //Make the first player passive
+    g.set_player_state_passive(g.get_player(1));
+
+    // Make player 1 and 2 overlap
+    g.get_player(1).set_x(get_x(g.get_player(0)));
+    g.get_player(1).set_y(get_y(g.get_player(0)));
+
+    const int first_player_index = get_collision_members(g)[1];
+    const int second_player_index = get_collision_members(g)[0];
+    const int loser_index = get_losing_player_index(g, first_player_index, second_player_index);
+
+    assert(loser_index == 1);
+
+    // After a tick, passive player does not shrink
+    const int inv_player_size_before =  get_nth_player_size(g, loser_index);
+    g.tick();
+    const int inv_player_size_after =  get_nth_player_size(g, loser_index);
     assert(inv_player_size_after == inv_player_size_before);
   }
   #endif
 
-  //#define FIX_ISSUE_463
+
+  #define FIX_ISSUE_463
   #ifdef FIX_ISSUE_463
-  // Players lose invulnerability after a short time
+  // Can get and set a player's passive state
+  {
+    game g;
+    player& p = g.get_player(0);
+
+    // Check default duration
+    assert(p.get_duration_passive() == 2000);
+
+    int new_duration = 200;
+    p.set_duration_passive(new_duration);
+    assert(p.get_duration_passive() == new_duration);
+  }
+
+  // Game cannot increment a player's passive timer unless it is passive
+  {
+    game g;
+    player& p = g.get_player(0);
+    assert(!is_passive(p));
+    assert(p.get_passive_timer() == 0);
+
+    g.increment_passive_timers();
+    assert(p.get_passive_timer() == 0);
+
+    int timer_before = p.get_passive_timer();
+    p.set_state_passive();
+    g.increment_passive_timers();
+    int timer_after = p.get_passive_timer();
+    int difference = timer_after - timer_before;
+    assert(difference == 1);
+  }
```

Comment:
Same comment as above for test 381; I don't think this adds anything that is not already (implicitly) required by the test I had written (below)